src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,42 @@
+package duke.task;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents a task that has a deadline.
+ */
+public class Deadline extends Task {
+    private final LocalDateTime by;
+
+    /**
+     * Creates a Deadline object.
+     * @param description The description of the deadline.
+     * @param by The deadline of the task.
+     */
+    public Deadline(String description, LocalDateTime by) {
+        super(description);
+        this.by = by;
+    }
+
+    /**
+     * Creates a Deadline object.
+     * @return The deadline of the task.
```

Comment:
Not all the Javadoc comments have an empty line between the description and parameter section, you could consider standardising the format.